src/python/pants/option/options.py

```diff
@@ -462,7 +456,9 @@ def get_fingerprintable_for_scope(
             for (_, kwargs) in sorted(parser.option_registrations_iter()):
                 if kwargs.get("recursive", False) and not kwargs.get("recursive_root", False):
                     continue  # We only need to fprint recursive options once.
-                if not kwargs.get(fingerprint_key, fingerprint_default):
+                if not kwargs.get("fingerprint", True):
```

Comment:
I don't fully understand the original, but this changes the default iiuc? Before, `invert` defaulted to `False`. This means `fingerprint_default` was also `False`. So, you were doing `kwargs.get("fingerprint", False)`. And now it's `True`.

Reply:
Before, we called this with either `fingerprint_key="fingerprint", invert=True`, or fingerprint_key="daemon", invert=False`. But that is exactly the kind of confusion I am trying to clear up here...

Reply:
Note that accidentally not passing `invert=True` previously would have had the effect of making all options non-fingerprintable by default (!!!!!!) But we clearly want options to be fingerprintable by default, and this code makes that much more obvious.

Reply:
Yes, in v1 fingerprinting was indeed off by default, and had to be enabled per-option. Now the converse is true.